internal/common/events/events.go

```diff
@@ -11,6 +11,173 @@ import (
     "github.com/go-openapi/strfmt"
 )
 
+//
+// Event generic_cluster
+//
+type GenericClusterEvent struct {
+    ClusterId strfmt.UUID
+    Message string
+    Severity string
+}
+
+var GenericClusterEventName string = "generic_cluster"
+
+func NewGenericClusterEvent(
+    clusterId strfmt.UUID,
+    message string,
+    severity string,
+) *GenericClusterEvent {
+    return &GenericClusterEvent{
+        ClusterId: clusterId,
+        Message: message,
+        Severity: severity,
+    }
+}
+
+func SendGenericClusterEvent(
+    ctx context.Context,
+    eventsHandler events.Sender,
+    clusterId strfmt.UUID,
+    message string,
+    severity string,) {
+    ev := NewGenericClusterEvent(
+        clusterId,
+        message,
+        severity,
+    )
+    eventsHandler.SendClusterEvent(ctx, ev)
+}
+
+func SendGenericClusterEventAtTime(
+    ctx context.Context,
+    eventsHandler events.Sender,
+    clusterId strfmt.UUID,
+    message string,
+    severity string,
+    eventTime time.Time) {
+    ev := NewGenericClusterEvent(
+        clusterId,
+        message,
+        severity,
+    )
+    eventsHandler.SendClusterEventAtTime(ctx, ev, eventTime)
+}
+
+func (e *GenericClusterEvent) GetName() string {
+    return "generic_cluster"
+}
+
+func (e *GenericClusterEvent) GetSeverity() string {
+    return e.Severity
+}
+
+func (e *GenericClusterEvent) GetClusterId() *strfmt.UUID {
```

Comment:
Why does this return a pointer? Then the calling code below immediately dereferences it (without a nil check)? Seems like it would be safer to just return `strfmt.UUID` and probably more efficient since it's just a string. WDYT?

Reply:
Addressed by PR #2438.